### ISSUE TITLE
Dark Text on Light Background is Rendered Weirdly

### ISSUE THREAD

**Issue by SnoutBug** [zed-industries/zed#19100]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I wondered why all light themes look terrible in Zed. They look washed out, and in general have less contrast than their dark theme counter-part. I looked around in other editors and found that the fonts in zed seem to be rendered differently compared to elsewhere on my system.
The issue is best shown by the screenshots below.

### Environment

<details><summary>System Details Report</summary>
<pre>
## Report details
- **Date generated:**                              2024-10-11 23:12:17
## Hardware Information:
- **Hardware Model:**                              Micro-Star International Co., Ltd MS-7B86
- **Memory:**                                      16,0 GiB
- **Processor:**                                   AMD Ryzen™ 5 2600 × 12
- **Graphics:**                                    AMD Radeon™ RX 7900 XT
## Software Information:
- **OS Name:**                                     Fedora Linux 40 (Workstation Edition)
- **OS Type:**                                     64-bit
- **GNOME Version:**                               46
- **Windowing System:**                            Wayland
- **Kernel Version:**                              Linux 6.10.12-200.fc40.x86_64
</pre>
</details>
<details><summary>Zed system specs</summary>
<pre>
Zed: v0.156.1 (Zed)
OS: Linux Wayland fedora 40
Memory: 15.5 GiB
Architecture: x86_64
GPU: AMD Radeon RX 7900 XT (RADV NAVI31) || radv || Mesa 24.1.7
</pre>
</details>

### If applicable, add mockups / screenshots to help explain present your vision of the feature

|Zed|Gnome Editor|
|-----|-----|
|![image](https://github.com/user-attachments/assets/d4c6a39e-10d0-4993-a168-2442e2b5bf35)|![image](https://github.com/user-attachments/assets/3499d4ba-701d-44c2-a1b8-1cbd6dcdc341)|
|![image](https://github.com/user-attachments/assets/d331de7a-c6b1-4b8d-ad8d-7e7250e7ef32)|![image](https://github.com/user-attachments/assets/2f7c5f38-322c-4e57-bce1-32a0edd63b08)|

Both Zed and the Gnome Editor use the same color scheme for dark and light mode. Notice how the light scheme in Zed looks drastically different (font appears thinner, more washed out and has seems to have less contrast) to the light theme in the Gnome Editor, or even the dark theme in Zed.

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by notpeter** [comment]
This is similar to:
- https://github.com/zed-industries/zed/issues/16401#issuecomment-2401395517

Can you see if the workaround suggested there improve things?

**Comment by SnoutBug** [comment]
The issue itself does not seem to be related, though it seems yioannides has had the same issue I face. The mentioned fix with the environment variables did not help, however. Changing the font itself also did not work.

**Comment by yioannides** [comment]
Yeah, the environment variable trick improves some of GNOME-wide aliasing issues, but Zed (literally) renders the font "wrong" in dark mode and I think it also affects both UI and monospace fonts used in Zed.

**Comment by yioannides** [comment]
UPDATE: While working on another unrelated issue, I removed the TTF version of Jetbrains and installed the OTF version instead, and the font surprisingly renders correctly:

![image](https://github.com/user-attachments/assets/4d9816e9-4692-488e-9e57-51734e7bf428)
![image](https://github.com/user-attachments/assets/6203ccbd-c410-42e4-8f63-e7d8b39925b5)

**Comment by SnoutBug** [comment]
Interestingly this did not help in my case. The light theme font still looks way thinner than the dark theme counterpart.
For me a workaround would be to increase the font-weight for the light theme by 100 relative to the corresponding dark theme.

**Comment by nirfse** [comment]
I experience the same issue on Arch Linux with SwayWM, and `zed-preview-bin` with any monospace font (`ttf-jetbrains-mono` in my case). Dark mode themes have somewhat contrast font rendering, while light themes produce thin squishy font lines, as if there's some anti-aliasing or char thickness rendering issue. Manually increasing `buffer_font_weight` adds some contrast, but font rendering is still very off (OP's screenshots are on point).

**Comment by scottpashley** [comment]
I'm using the JetBrains mono fonts, I removed them all and then re-installed only the Semibold variant, which is what I use everywhere anyway. I went with the OTF version as suggested by @yioannides and now it looks as it should

**Comment by rijenkii** [comment]
Same issue with the *default fonts*, makes zed basically unusable on light mode:
![image](https://github.com/user-attachments/assets/85eb3a1e-942f-4445-bd5e-0d41105bd759)

<details><summary>specs</summary>

```
Zed: v0.163.2 (Zed)
OS: Linux Wayland fedora 41
Memory: 30.7 GiB
Architecture: x86_64
GPU: AMD Radeon Graphics (RADV RENOIR) || radv || Mesa 24.2.6
```

</details>

**Comment by iamnbutler** [comment]
@as-cii has an experimental PR up for this – if someone who is familiar with running zed from source wants to try checking out this PR and see if it helps that could help us make some decisions around if it makes sense to merge it or not. 

PR: https://github.com/zed-industries/zed/pull/20506

**Comment by rijenkii** [comment]
Are you sure that PR will affect this issue? PR mentions specifically macOS in the title, but OP (and me) are running linux wayland.

**Comment by iamnbutler** [comment]
Hmm... I'm not familiar enough with the mechanics of our text system to know if that PR will impact linux or not. I'll let @as-cii chime in.

**Comment by katyo** [comment]
Light themes still looks weird on Linux! Any suggestions?
Zed 0.176.3

**Comment by PauloVBettio** [comment]
**Default font weight**
![Image](https://github.com/user-attachments/assets/0023f375-2af7-4da2-8a1a-936b743ac47b)

**Font weight set to 600**
![Image](https://github.com/user-attachments/assets/6fff910a-a24e-478e-8c92-a9aad686f993)

Please, open the images on a new tab. Github is scaling down the images to fit the layout.

Overlaying both dark and light themes you can really see a difference in contrast.
https://github.com/user-attachments/assets/9c55a731-cbc9-4f3f-9240-054df9d6e5b0

**Comment by notpeter** [comment]
I'm going to go ahead and close in favor of:
- https://github.com/zed-industries/zed/issues/7992

We are working on prioritizing/improving font rendering on lower DPI displays and intend to fix this at the same time.

Please 👍 and subscribe to that issue  be notified of progress on that issue. Thanks for reporting.
